pkg/build/generator/generator.go

```diff
@@ -210,6 +259,19 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from *kap
 
 	glog.V(4).Infof("Resolving ImageStreamReference %s of Kind %s in namespace %s", from.Name, from.Kind, namespace)
 	switch from.Kind {
+	case "ImageStream":
+		// NOTE: The 'ImageStream' reference should be used only for the 'output' image
```

Comment:
Looking at [build's validation](https://github.com/openshift/origin/blob/master/pkg/build/api/validation/validation.go#L207) I don't see Kind validation for From, there's some for [To](https://github.com/openshift/origin/blob/master/pkg/build/api/validation/validation.go#L135) but I'll be changing that. One way or the other, I'll wait with my PR for your to get in and will fix it appropriately.